src/hooks/usePagination.tsx

```diff
@@ -0,0 +1,48 @@
+export const usePagination = (
+  currentPage: number,
+  postsTotal: number | undefined
+) => {
+  const pagesAmount = Math.ceil(Number(postsTotal) / 8);
+  const allPages = currentPage;
+  let buttonList = [];
```

Comment:
You should provide a type for this array. BTW, do you need a `List` prefix? Can't it just me `let buttons = []`?